browser/ipfs/content_browser_client_helper.cc

```diff
@@ -132,13 +132,15 @@ bool HandleIPFSURLReverseRewrite(GURL* url,
     return false;
   GURL::Replacements scheme_replacements;
   GURL::Replacements host_replacements;
+  auto decoded_host = ipfs::DecodeSingleLabelForm(
```

Comment:
ipns_pos could be string::npos here, also could we move it before IsValidCIDOrDomain and reuse this value?